src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -82,6 +84,10 @@ private static bool InUnsafeMethod(Symbol symbol)
         private Dictionary<LocalSymbol, (uint RefEscapeScope, uint ValEscapeScope)>? _localEscapeScopes;
         private Dictionary<BoundValuePlaceholderBase, uint>? _placeholderScopes;
```

Comment:
>uint [](http://example.com/codeflow?start=54&length=4)

I thought the idea for the long term fix was to substitute placeholders with expressions they represent.

Reply:
We could store the replacement expression in the dictionary rather than the escape scope, and evaluate the escape scope of the replacement lazily, when it is used, rather than eagerly, when the placeholder is registered. However, that would mean ensuring `GetValEscape()` does not rely on mutable state in the visitor such as `_localScopeDepth` or `_inUnsafeRegion`. Given that, evaluating the escape scope eagerly seemed like a simpler alternative.